### DISCUSSION TITLE
how can I add a atmospheric deposition into fabm

### DISCUSSION THREAD

**Discussion by chevyyang** [fabm-model/fabm#43] [Q&A] [answered]
Hi, everyone~~ I am tring to couple ERSEM-FABM-FVCOM to simulate the phytoplankton distribution. I want to add a atmospheric deposition to the couppled model. I add the precipitation file to the fvcom and add biological parameters in that nc file. but I found the coupled model wont read the nutrients in the precipitation nc file. What should I do? How can I add the N3_n amd N1_p to the coupled model.

for the L4 example, the yaml file has the content like:
  N_sfl_denit:
    model: external_surface_flux
    coupling:
      target: N3/n
      flux: ben_nit/G4n_sms_tot
  N_sfl_burial:
    model: external_surface_flux
    coupling:
      target: N3/n
      flux: Q17/n_sms_tot
  P_sfl_burial:
    model: external_surface_flux
    coupling:
      target: N1/p
      flux: Q17/p_sms_tot

are they about atmospheric deposition? and what is the mean " flux: Q17/p_sms_tot" ? give the parameter name for n1p in atmospheric deposition nc file ? where can I put the atmospheric deposition nc file name for the couppled model to recognize the dynamic forcing file? what is the paramter names for n1p or n3n in atmospheric deposition file? usually the unit should be umol /m2/s , am I right?  thank you so much for your help and time in addvance.

**Comment by jornbr** [comment, marked as answer]
Hi, these questions are quite specific to the FVCOM-FABM coupler and ERSEM and you may therefore want to follow up with the people behind those. But for what it's worth: the yaml snippet you show creates surface fluxes for `N3/n` (nitrate, 2x) and for `N1/p` (phosphate). But instead of speciyfing them externally, the surface fluxes are instead set equal to the few non-conservative fluxes in the model, namely the denitrification rate (`ben_nit/G4n_sms_tot`) and burial rates of nitrogen and phosphorus (`Q17/n_sms_tot`, `Q17/p_sms_tot`). This is done to ensure the total bioavailable amounts of nitrogen and phosphorus in the system stay constant over time.

If you want to add atmospheric deposition from file, the typical way to do this is to omit the `coupling/flux` part of the snippet above. For instance:

```
N_dep:
  model: external_surface_flux
  coupling:
    target: N3/n
```

That would leave `N_dep/flux` as unresolved model dependency that you have to provide via FVCOM. I believe that it typcially done vie a file called `fabm_input.nml`, but this is specific to FVCOM, not FABM.

You can also prescribe a constant surface flux (constant in both time and space) via `fabm.yaml` alone:

```
N_dep:
  model: constant_surface_flux
  parameters:
    flux: 6.94e-07   # state variable units * m s-1, which here means mmol N m-2 s-1
  coupling:
    target: N3/n
```
